lib/LMSManagers/LMSHelpdeskManager.php

```diff
@@ -1718,8 +1718,13 @@ public function TicketChange($ticketid, array $props)
             $props['customerid'] = null;
         }
 
-        if (isset($props['categories'])) {
-            $ticket['categories'] = empty($ticket['categories']) ? array() : array_keys($ticket['categories']);
+        $ticket['categories'] = is_array($ticket['categories']) ? Utils::array_column($ticket['categories'], 'id') : array();
+        $props['categories'] = is_array($props['categories']) ? array_values($props['categories']) : array();
+
+        if (!Utils::arrays_equal($ticket['categories'], $props['categories'])) {
+            $ticket['categories'] = empty($ticket['categories']) ? array() : $ticket['categories'];
+            $props['categories'] = empty($props['categories']) ? array() : $props['categories'];
+
```

Comment:
Po ostatnich zmianach to nadal potrzebne?

Reply:
Tak - a czemu masz wątpliwości, że nie?

Reply:
Nie widzę tutaj miejsca na uproszczenie kodu a Ty?

Reply:
@chilek

Reply:
@chilek

Reply:
Po co sprawdzasz czy `empty(...)`, gdy już wwcześniej zadbałeś, żeby było tu na pewno `array()`?

Reply:
Tak jest potwierdzam - https://github.com/chilek/lms/pull/2150/commits/02c07789845f008ca42af63ac7d04a1159d7630d

W ogóle to natchnąłeś mnie - to da się jeszcze mocniej uprościć. Ale to puszczę już w innym PR.

Reply:
Także proszę o squash&merge.